File: src/main/java/cmc/mody/auth/presentation/dto/TokenDto.java
package cmc.mody.auth.presentation.dto;

public record TokenDto(
    Long id,
    String accessToken,
    String refreshToken,
    boolean personalInfoCompleted,
    boolean mainAccessible,
    boolean groupOnboardingCompleted
) {
    public static TokenDto of(Long id, String accessToken, String refreshToken) {
        return of(id, accessToken, refreshToken, false, false, false);
    }

    public static TokenDto of(
        Long id,
        String accessToken,
        String refreshToken,
        boolean personalInfoCompleted
    ) {
        return of(id, accessToken, refreshToken, personalInfoCompleted, false, false);
    }

    public static TokenDto of(
        Long id,
        String accessToken,
        String refreshToken,
        boolean personalInfoCompleted,
        boolean mainAccessible
    ) {
        return of(id, accessToken, refreshToken, personalInfoCompleted, mainAccessible, false);
    }

    public static TokenDto of(
        Long id,
        String accessToken,
        String refreshToken,
        boolean personalInfoCompleted,
        boolean mainAccessible,
        boolean groupOnboardingCompleted
    ) {
        return new TokenDto(
            id,
            accessToken,
            refreshToken,
            personalInfoCompleted,
            mainAccessible,
            groupOnboardingCompleted
        );
    }
}

File: src/main/kotlin/cmc/mody/auth/presentation/dto/TokenDto.kt
package cmc.mody.auth.presentation.dto

data class TokenDto(
    val id: Long,
    val accessToken: String,
    val refreshToken: String,
    val isNewMember: Boolean
) {
    companion object {
        fun of(
            id: Long,
            accessToken: String,
            refreshToken: String,
            isNewMember: Boolean = false
        ): TokenDto =
            TokenDto(
                id = id,
                accessToken = accessToken,
                refreshToken = refreshToken,
                isNewMember = isNewMember
            )
    }
}
